packages/SwingSet/src/vats/timer/vat-timer.js

```diff
@@ -1,98 +1,977 @@
+   * wakeup time) at 'delay' time units in the future.
+   *
+   * @param {RelativeTime} delay
+   * @param {CancelToken} [cancelToken]
+   * @returns { Promise<Timestamp> }
+   */
+  function addDelay(delay, cancelToken = undefined) {
+    delay = fromRelativeTime(delay);
+    assert(delay >= 0n, 'delay must not be negative');
+    const now = getNow();
+    if (delay === 0n) {
+      return Promise.resolve(toTimestamp(now));
+    }
+    const when = now + delay;
+    return wakeAtInternal(when, cancelToken);
+  }
+
+  /**
+   * cancel(token): Cancel an outstanding one-shot, or a Notifier
+   * (outstanding or idle), or new-style repeater (not `makeRepeater`,
+   * which has .disable). For things that return Promises, the Promise
+   * is rejected with Error('TimerCancelled').
+   *
+   * @param {CancelToken} cancelToken
+   */
+  function cancel(cancelToken) {
+    // silently ignore multiple cancels and bogus token
+    if (cancels.has(cancelToken)) {
+      cancels.get(cancelToken).forEach(thing => thing.cancel());
+    }
+  }
+
+  /**
+   * Internal function to register a handler, which will be invoked as
+   * handler.wake(scheduledTime) at the earliest non-past instance of
+   * `start + k*interval`. When the wake() result promise
+   * fulfills, the repeater will be rescheduled for the next such
+   * instance (there may be gaps). If that promise rejects, the
+   * repeater will be cancelled. The repeater can also be cancelled by
+   * providing `cancelToken` and calling
+   * `E(timerService).cancel(cancelToken)`.
+   *
+   * @param {TimestampValue} start
+   * @param {RelativeTimeValue} interval
+   * @param {Handler} handler
+   * @param {CancelToken} [cancelToken]
+   */
+  function repeat(start, interval, handler, cancelToken) {
+    assert.typeof(start, 'bigint');
+    assert.typeof(interval, 'bigint');
+    assert(interval > 0n, 'interval must be positive');
+    const event = makeRepeaterEvent(start, interval, handler, cancelToken);
+    // computes first wakeup, inserts into schedule, updates alarm. If
+    // start has passed already, fires immediately.
+    event.scheduleYourself();
+  }
+
+  // --- Repeaters: legacy "distinct Repeater object" API ---
+
+  // The durable Repeater object is built from (delay, interval)
+  // arguments which requests a wakeup at the earliest non-past
+  // instance of `now + delay + k*interval`. The returned object
+  // provides {schedule, disable} methods. We build an Event from it.
+
+  function initRepeater(delay, interval) {
+    // first wakeup at now+delay, then now+delay+k*interval
+    delay = fromRelativeTime(delay);
+    assert(delay >= 0n, 'delay must be non-negative');
+    interval = fromRelativeTime(interval);
+    assert(interval > 0n, 'interval must be nonzero');
+    const start = getNow() + delay;
+    const active = false;
+    return { start, interval, active };
+  }
+  const repeaterFacets = {
+    cancel: {}, // internal CancelToken
```

Comment:
Aye. Do you know of any way to maintain the visual conciseness of:

```js
  const timerService = vivifySingleton(baggage, 'timerService', {
    getCurrentTimestamp,
    setWakeup, // one-shot with handler (absolute)
    wakeAt, // one-shot with Promise (absolute)
    delay: addDelay, // one-shot with Promise (relative)
    makeRepeater, // repeater with Repeater control object (old)
    repeatAfter, // repeater without control object
    makeNotifier, // Notifier
    cancel, // cancel setWakeup/wakeAt/delay/repeat
    getClock: () => timerClock,
    getTimerBrand: () => timerBrand,
  });
```

given that IDE interpretation constraint? I'd find the following less readable, although maybe that's the only way to avoid the problem:
```js
  const timerService = vivifySingleton(baggage, 'timerService', {
    getCurrentTimestamp,
    // one-shot with handler (absolute)
    setWakeup,
    // one-shot with Promise (absolute)
    wakeAt,
    // one-shot with Promise (relative)
    delay: addDelay,
    // repeater with Repeater control object (old)
    makeRepeater,
    // repeater without control object
    repeatAfter,
    // Notifier
    makeNotifier,
    // cancel setWakeup/wakeAt/delay/repeat
    cancel,
    getClock: () => timerClock,
    getTimerBrand: () => timerBrand,
  });
```

Reply:
This is as concise. The use of `/* */` comments means that typescript doesn't pick them up, but perhaps that's what you want here.

``` js
  const timerService = vivifySingleton(baggage, 'timerService', {
    getCurrentTimestamp,
    setWakeup /* one-shot with handler (absolute) */,
    wakeAt /* one-shot with Promise (absolute) */,
    delay: addDelay /* one-shot with Promise (relative) */,
    makeRepeater /* repeater with Repeater control object (old) */,
    repeatAfter /* repeater without control object */,
    makeNotifier /* Notifier */,
    cancel /* cancel setWakeup/wakeAt/delay/repeat */,
    getClock: () => timerClock,
    getTimerBrand: () => timerBrand,
  });
```

Reply:
ah! I hadn't even thought about `/**/` being treated differently. Applied, thanks.